ui/components/app/permission-connect-header/permission-connect-header.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  AlignItems,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+  Display,
+  BlockSize,
+  FontWeight,
+  FlexDirection,
+  BackgroundColor,
+} from '../../../helpers/constants/design-system';
+import {
+  IconSize,
+  Text,
+  Box,
+  AvatarFavicon,
+  AvatarBase,
+} from '../../component-library';
+
+const PermissionConnectHeader = ({ origin, iconUrl }) => {
+  const transformOriginToTitle = (rawOrigin) => {
+    try {
+      const url = new URL(rawOrigin);
+      const parts = url.hostname.split('.');
+      return parts.slice(-2).join('.');
```

Comment:
Are we cool with removing subdomain(s) here?

Reply:
Yes. This was design requirement. First we show the main domain, then below we show the full URL with subdomain.
![Screenshot 2024-04-23 at 22 35 50](https://github.com/MetaMask/metamask-extension/assets/13301024/d338d5bb-98ca-4bfd-8dea-d36fc34a57fb)